DA Layer Submission Failures should result in incremental backoff

Currently DA Layer submissions are retried immediately. This means that all 10 attempts may occur during a single transient delay. We should configure the retries to use an incremental backoff. 

https://github.com/cenkalti/backoff

Example Error Log
```
9:39PM ERR DA layer submission failed error="Post \"http://127.0.0.1:26658/submit_pfd\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)" attempt=1 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=2 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=3 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=4 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=5 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=6 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=7 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=8 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=9 module=BlockManager
9:39PM ERR DA layer submission failed error= attempt=10 module=BlockManager
panic: Failed to submit block to DA layer!
```

--- a/block/manager.go
+++ b/block/manager.go
@@ -29,7 +29,10 @@ const defaultDABlockTime = 30 * time.Second
 
 // maxSubmitAttempts defines how many times Optimint will re-try to publish block to DA layer.
 // This is temporary solution. It will be removed in future versions.
-const maxSubmitAttempts = 10
+const maxSubmitAttempts = 30
+
+// initialBackoff defines initial value for block submission backoff
+var initialBackoff = 100 * time.Millisecond
 
 type newBlockEvent struct {
 	block    *types.Block
@@ -395,6 +398,7 @@ func (m *Manager) submitBlockToDA(ctx context.Context, block *types.Block) error
 	m.logger.Debug("submitting block to DA layer", "height", block.Header.Height)
 
 	submitted := false
+	backoff := initialBackoff
 	for attempt := 1; ctx.Err() == nil && !submitted && attempt <= maxSubmitAttempts; attempt++ {
 		res := m.dalc.SubmitBlock(block)
 		if res.Code == da.StatusSuccess {
@@ -402,6 +406,8 @@ func (m *Manager) submitBlockToDA(ctx context.Context, block *types.Block) error
 			submitted = true
 		} else {
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
+			time.Sleep(backoff)
+			backoff = m.exponentialBackoff(backoff)
 		}
 	}
 
@@ -413,6 +419,14 @@ func (m *Manager) submitBlockToDA(ctx context.Context, block *types.Block) error
 	m.HeaderOutCh <- &block.Header
 
 	return nil
+}
+
+func (m *Manager) exponentialBackoff(backoff time.Duration) time.Duration {
+	backoff *= 2
+	if backoff > m.conf.DABlockTime {
+		backoff = m.conf.DABlockTime
+	}
+	return backoff
 }
 
 func (m *Manager) publishHeader(block *types.Block) {

--- a/da/celestia/celestia.go
+++ b/da/celestia/celestia.go
@@ -3,6 +3,7 @@ package celestia
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"time"
 
 	"github.com/gogo/protobuf/proto"
@@ -81,7 +82,7 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.ResultS
 		return da.ResultSubmitBlock{
 			DAResult: da.DAResult{
 				Code:    da.StatusError,
-				Message: txResponse.RawLog,
+				Message: fmt.Sprintf("Codespace: '%s', Code: %d, Message: %s", txResponse.Codespace, txResponse.Code, txResponse.RawLog),
 			},
 		}
 	}

